core/src/main/java/org/ehcache/config/writebehind/DefaultWriteBehindConfiguration.java

```diff
@@ -25,94 +25,57 @@
-  }
-  
-  @Override
-  public int getMinWriteDelay() {
-    return minWriteDelay;
-  }
-  
-  @Override
-  public int getMaxWriteDelay() {
-    return maxWriteDelay;
-  }
-  
-  @Override
-  public boolean isWriteCoalescing() {
-    return writeCoalescing;
-  }
-  
-  @Override
-  public int getWriteBatchSize() {
-    return writeBatchSize;
+  public DefaultWriteBehindConfiguration(BatchingConfiguration batchingConfig) {
+    this.batchingConfig = batchingConfig;
   }
   
   @Override
-  public int getWriteBehindConcurrency() {
+  public int getConcurrency() {
     return writeBehindConcurrency;
   }
   
   @Override
-  public int getWriteBehindMaxQueueSize() {
+  public int getMaxQueueSize() {
     return writeBehindMaxQueueSize;
   }
 
-  public void setMinWriteDelay(int minWriteDelay) {
-    if (minWriteDelay < 0) {
-      throw new IllegalArgumentException("Minimum write delay seconds cannot be less than 0");
-    } else if (minWriteDelay > maxWriteDelay) {
-      throw new IllegalArgumentException("Minimum write delay (" + minWriteDelay +
-                                         ") must be smaller than or equal to maximum write delay (" + maxWriteDelay + ")");
-    }
-    this.minWriteDelay = minWriteDelay;
-  }
-
-  public void setMaxWriteDelay(int maxWriteDelay) {
-    if (maxWriteDelay < 0) {
-      throw new IllegalArgumentException("Maximum write delay cannot be less than 1");
-    } else if (maxWriteDelay < minWriteDelay) {
-      throw new IllegalArgumentException("Maximum write delay (" + maxWriteDelay +
-                                         ") must be larger than or equal to minimum write delay (" + minWriteDelay + ")");
-    }
-    this.maxWriteDelay = maxWriteDelay;
-  }
-
-  public void setWriteCoalescing(boolean writeCoalescing) {
-    this.writeCoalescing = writeCoalescing;
+  @Override
+  public String getThreadPoolAlias() {
+    return executorAlias;
   }
 
-  public void setWriteBatchSize(int writeBatchSize) {
-    if(writeBatchSize < 1) {
-      throw new IllegalArgumentException("Batchsize cannot be less than 1.");
-    }
-    this.writeBatchSize = writeBatchSize;
+  @Override
+  public BatchingConfiguration getBatchingConfiguration() {
+    return batchingConfig;
   }
 
-  public void setWriteBehindConcurrency(int writeBehindConcurrency) {
+  public void setConcurrency(int writeBehindConcurrency) {
     if(writeBehindConcurrency < 1) {
       throw new IllegalArgumentException("Concurrency Level cannot be less than 1.");
     }
     this.writeBehindConcurrency = writeBehindConcurrency;
   }
 
-  public void setWriteBehindMaxQueueSize(int writeBehindMaxQueueSize) {
+  public void setMaxQueueSize(int writeBehindMaxQueueSize) {
     if(writeBehindMaxQueueSize < 1) {
       throw new IllegalArgumentException("WriteBehind queue size cannot be less than 1.");
     }
     this.writeBehindMaxQueueSize = writeBehindMaxQueueSize;
   }
 
+  public void setExecutorAlias(String executorAlias) {
```

Comment:
Configuration classes should be immutable. While I know this was not the case before, I believe it is worth making it so.